compiler-utils/src/main/java/com/squareup/anvil/compiler/internal/reference/ClassReference.kt

```diff
@@ -306,6 +306,28 @@ public fun FqName.toClassReference(module: ModuleDescriptor): ClassReference {
 public fun ClassReference.isAnnotatedWith(fqName: FqName): Boolean =
   annotations.any { it.fqName == fqName }
 
+@ExperimentalAnvilApi
+public fun ClassReference.annotationOrNull(
+  annotationName: FqName,
+  scopeName: FqName? = null
+): AnnotationReference? {
+  return annotations.singleOrNull { it.fqName == annotationName }
+    ?.takeIf { scopeName == null || it.scopeOrNull()?.fqName == scopeName }
+}
+
+@ExperimentalAnvilApi
+public fun ClassReference.annotation(
+  annotationName: FqName,
+  scopeName: FqName? = null
+): AnnotationReference {
```

Comment:
I really don't like these extension. Once we have repeatable annotations they no longer work. We shouldn't move the code into the wrong direction.

Reply:
:+1: let's talk offline about what we expect things to look like once we have repeatable annotations in and see if this can be done in a more future-proof way. I just found myself writing this code manually many many times during this PR and felt like it was better to consolidate the logic somewhere, but I'm down to make changes.